sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserver.java

```diff
@@ -39,6 +43,7 @@ public void onEvent(int eventType, @Nullable String relativePath) {
 
     // TODO: Only some event types should be pass through?
 
-    envelopeSender.processEnvelopeFile(this.rootPath + File.separator + relativePath);
+    final CachedEnvelopeHint hint = new CachedEnvelopeHint(timeout, logger);
```

Comment:
Wasn't the lack of this hint that made events captured by this class get the scope applied?

Reply:
`Cached` means it's being read from disk, but still doesn't mean we should or should not apply scope's data.
App. is alive and it should apply scope's data, that's why the combination of being Cached + ApplyScopeData.
this Hint could be an internal class of EnvelopeFileObserver

Reply:
What does `Cached` alone control?

Reply:
`DefaultAndroidEventProcessor` needs it to not overwrite data (eg proguard Id, if App has restarted, we don't want to take the new ID) - similar to scope's data.

`MainEventProcessor` and `SentryClient` as you already know, apply or not scope's data.

`AsyncConnection` needs it to set a `NoOp` for `IEventCache` and `IEnvelopeCache`, so it's not needed to cache it again.

Reply:
OK so we'll keep both: 
*  `Cached` don't store this on disk, it's already from disk
*  `ApplyScope`: apply scope 

that makes sense, thanks for clarifying